api.planx.uk/modules/analytics/metabase/shared/client.ts

```diff
@@ -0,0 +1,155 @@
+import axios from "axios";
+import type {
+  AxiosInstance,
+  AxiosError,
+  AxiosRequestConfig,
+  InternalAxiosRequestConfig,
+} from "axios";
+import dotenv from "dotenv";
+
+dotenv.config();
+
+// Custom error class for Metabase-specific errors
+export class MetabaseError extends Error {
+  constructor(
+    message: string,
+    public statusCode?: number,
+    public response?: unknown,
+  ) {
+    super(message);
+    this.name = "MetabaseError";
+  }
+}
+
+interface MetabaseConfig {
+  baseURL: string;
+  apiKey: string;
+  timeout?: number;
+  retries?: number;
+}
+
+// Validate environment variables
+const validateConfig = (): MetabaseConfig => {
+  const baseURL = process.env.METABASE_BASE_URL;
+  const apiKey = process.env.METABASE_API_KEY;
+
+  if (!baseURL) {
+    throw new Error(
+      "METABASE_BASE_URL is not defined in environment variables",
+    );
+  }
+  if (!apiKey) {
+    throw new Error("METABASE_API_KEY is not defined in environment variables");
+  }
```

Comment:
nit: We generally use a pattern of calling `assert()` for env vars (which also accepts a second message argument) across the code base.

I think it's worth keeping this pattern as I'd really like to do a find and replace sometime and replace it with an approach like this which would be super nice! https://www.youtube.com/watch?v=q1im-hMlKhM 🧙‍♂️